tests/test_pytomorrowio.py

```diff
@@ -86,16 +82,60 @@ async def _test_capture_request_and_response():
         assert False
 
 
-async def test_timelines_hourly_good():
-    with create_call_api_mock() as mock:
-        mock.return_value = load_json("timelines_hourly_good.json")
+def set_mock_return_value(mock: Mock, return_value):
+    mock.return_value = return_value
+    return None  # return None so caller can chain in lambda func
 
-        api = TomorrowioV4("bogus_api_key", *GPS_COORD)
-        available_fields = api.available_fields(
-            ONE_HOUR, [TYPE_POLLEN, TYPE_PRECIPITATION, TYPE_WEATHER]
-        )
-        res = await api.forecast_hourly(available_fields)
-        mock.assert_called_once()
+
+@patch.object(TomorrowioV4, "rate_limits", new_callable=PropertyMock)
+@patch.object(TomorrowioV4, "_call_api")
+async def test_rate_limits(call_api_mock: Mock, rate_limits_mock: Mock):
+    rate_limits_return_value = {
+        "RateLimit-Limit": "3",
+        "RateLimit-Remaining": "2",
+        "RateLimit-Reset": "1",
+        "X-RateLimit-Limit-Day": "500",
+        "X-RateLimit-Limit-Hour": "25",
+        "X-RateLimit-Limit-Second": "3",
+        "X-RateLimit-Remaining-Day": "447",
+        "X-RateLimit-Remaining-Hour": "24",
+        "X-RateLimit-Remaining-Second": "2",
+    }
+
+    call_api_mock.side_effect = lambda _: set_mock_return_value(
+        rate_limits_mock, rate_limits_return_value
+    ) or load_json("timelines_hourly_good.json")
```

Comment:
`side_effect` property allows a callable value where `return_value` doesn't. I used that to kludge in a change to the `rate_limits` property when a forecast is retrieved.